controller/helper/process-survey-instances.js

```diff
@@ -12,17 +12,41 @@ const moment = require('moment');
  * @returns {Object} Complience chart data
  */
 function processSurveyInstances (surveys) {
+    const filterSurveyByState = surveys.filter((survey) => {
+        return survey.state === 'completed';
+    });
+
     return {
         // Using UTC as the date gets modified to the local time (GMT in this case) when UTC not used.
-        labels: surveys.map((survey) => {
-            return moment(survey.startTime).utc().format('MM/DD/YYYY HH:mm');
-        }),
-        datasets: pickTimeLeft(surveys)
+        labels: pickDates(filterSurveyByState),
+        datasets: pickTimeLeft(filterSurveyByState)
     };
 }
 
 /**
  * Takes in a Survey Instances and get the % time left to be shown on complience chart
+ * @param {Object} surveys - list of survey instances
+ * @returns {Object} processed list of datetimes
+ */
+function pickDates (surveys) {
+    const dates = surveys.map((survey) => {
+        return moment(new Date(survey.startTime).toISOString()).utc().format('MM/DD/YYYY HH:mm');
+    });
+
+    const lastDate = surveys.map((survey) => {
+        return moment(new Date(survey.dateCompleted).toISOString()).utc().format('MM/DD/YYYY HH:mm');
+    });
+
+    const missingValue = -1;
+
+    if (dates.indexOf(lastDate) <= missingValue) {
+        dates.push(lastDate);
+    }
+    return dates;
```

Comment:
okay I'm still confused on what lines 36-44 are doing.
Now it looks to be:

1. formatting all the dates and storing array of dates `to lastDate` (lines 36-38)
2. checking if any `Date` in array `dates` is an array of dates (line 42)
3. if there is not one, append the array as the last element (line 43)
    - which will result in an array that looks something like `[Date, Date, Date, [Date, Date, Date]]`

again what is the intended outcome from `pickDates`?